onmt/translate/translator.py

```diff
@@ -58,37 +56,16 @@ def build_translator(opt, report_score=True, logger=None, out_file=None):
             report_score=report_score,
             logger=logger,
         )
-    return translator
-
-
-def max_tok_len(new, count, sofar):
-    """
-    In token batching scheme, the number of sequences is limited
-    such that the total number of src/tgt tokens (including padding)
-    in a batch <= batch_size
-    """
-    # Maintains the longest src and tgt length in the current batch
-    global max_src_in_batch  # this is a hack
-    # Reset current longest length at a new batch (count=1)
-    if count == 1:
-        max_src_in_batch = 0
-        # max_tgt_in_batch = 0
-    # Src: [<bos> w1 ... wN <eos>]
-    max_src_in_batch = max(max_src_in_batch, len(new.src[0]) + 2)
-    # Tgt: [w1 ... wM <eos>]
-    src_elements = count * max_src_in_batch
-    return src_elements
+    return translator, vocabs
```

Comment:
Do we really need to return `vocabs`, and break the signature of `build_translator`, since we can access `translator.vocabs`?

Reply:
good point you are correct. will change this.